src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -638,8 +638,6 @@ void GameObject::Update(uint32 diff)
                 return;
             }
 
-            SetLootState(GO_READY);
```

Comment:
Had to move SetLootState *after* flags reset, so the script can be properly notified.

Without this change, `SetLootState` executed the SAI, then the template flags were applied, leaving you with a non usable lever for the rest of the fight.